core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java

```diff
@@ -114,87 +125,50 @@ protected SwaggerBundleConfiguration getSwaggerBundleConfiguration(StreamRegistr
 
     @Override
     public void run(final StreamRegistryConfiguration configuration, final Environment environment) {
-        Properties producerProperties = new Properties();
-        producerProperties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class);
-        producerProperties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class);
+        /*
+         * Step 1 - initialize managed beans
+         */
 
-        configuration.getKafkaProducerConfig().getKafkaProducerProperties().forEach(producerProperties::put);
+        ManagedKStreams managedKStreams = createManagedKStreams(configuration);
 
-        Properties kstreamsProperties = new Properties();
-        configuration.getKafkaStreamsConfig().getKstreamsProperties().forEach(kstreamsProperties::put);
-        kstreamsProperties.put(ROCKSDB_CONFIG_SETTER_CLASS_CONFIG, CustomRocksDBConfig.class);
-        TopicsConfig topicsConfig = configuration.getTopicsConfig();
+        ManagedInfraManager managedInfraManager = createManagedInfraManager(configuration);
+        Preconditions.checkState(managedInfraManager != null, "managedInfraManager cannot be null.");
 
-        ManagedKafkaProducer managedProducer = new ManagedKafkaProducer(producerProperties, topicsConfig);
-        ManagedKStreams managedKStreams = new ManagedKStreams(kstreamsProperties, topicsConfig);
-
-        InfraManagerConfig infraManagerConfig = configuration.getInfraManagerConfig();
-        String infraManagerClassName = infraManagerConfig.getClassName();
+        ManagedKafkaProducer managedKafkaProducer = createManagedKafkaProducer(configuration);
 
-        InfraManager infraManager = null;
-        if (infraManagerClassName != null && !infraManagerClassName.isEmpty()) {
-            try {
-                infraManager = Utils.newInstance(infraManagerClassName, InfraManager.class);
-                infraManager.configure(infraManagerConfig.getConfig());
-            } catch (Exception e) {
-                log.error("Error loading/configuring Infra Manager Class", e);
-                throw new IllegalStateException("Could not load/configure Infra Manager class", e);
-            }
-        }
-        ManagedInfraManager managedInfraManager = new ManagedInfraManager(infraManager);
+        /*
+         * Step 2 - initialize the DAO's
+         */
```

Comment:
**_Note:_** - While re-factoring `createManagedKStreams`, `createManagedInfraManager` and `createManagedKafkaProducer` sensed an opportunity to centralize DAO creation in a separate method. Should this looks to be an over kill then open to make Step `2` inline and also for the other steps below.